rom/laghos_rom.cpp

```diff
@@ -13,6 +13,110 @@
 
 using namespace std;
 
+
+void DMD_Sampler::SampleSolution(const double t, const double dt, Vector const& S)
+{
+    SetStateVariables(S);
+    SetStateVariableRates(dt);
+
+    Vector dSdt;
+    if (!sns)
+    {
+        dSdt.SetSize(S.Size());
+        lhoper->Mult(S, dSdt);
+    }
+
+    if (!useXV)
+    {
+        if (rank == 0)
+        {
+            cout << "X taking sample at t " << t << endl;
+        }
+
+        if (offsetInit)
+        {
+            for (int i=0; i<tH1size; ++i)
+            {
+                Xdiff[i] = X[i] - (*initX)(i);
+            }
+
+            if (t >= tbegin) dmd_X->takeSample(Xdiff.GetData(), t);
+        }
+        else
+        {
+            if (t >= tbegin) dmd_X->takeSample(X.GetData(), t);
+        }
+    }
+
+    if (!useVX)
+    {
+        if (rank == 0)
+        {
+            cout << "V taking sample at t " << t << endl;
+        }
+
+        if (offsetInit && Voffset)
+        {
+            for (int i=0; i<tH1size; ++i)
+            {
+                Xdiff[i] = V[i] - (*initV)(i);
+            }
+
+            if (t >= tbegin) dmd_V->takeSample(Xdiff.GetData(), t);
+        }
+        else
+        {
+            if (t >= tbegin) dmd_V->takeSample(V.GetData(), t);
+        }
+    }
+
+    if (!sns)
+    {
+        MFEM_VERIFY(gfH1.Size() == H1size, "");
+        for (int i=0; i<H1size; ++i)
+            gfH1[i] = dSdt[H1size + i];  // Fv
+
+        gfH1.GetTrueDofs(Xdiff);
```

Comment:
This can be removed, as well as the computation of `dSdt`, since it is not sampled for DMD. Also for `Fe` below.

Reply:
Removed.

Reply:
The computation of `dSdt` is still there.